src/Coordination/KeeperStorage.cpp

```diff
@@ -1097,6 +1098,14 @@ struct KeeperStorageGetRequestProcessor final : public KeeperStorageRequestProce
         Coordination::ZooKeeperGetResponse & response = dynamic_cast<Coordination::ZooKeeperGetResponse &>(*response_ptr);
         Coordination::ZooKeeperGetRequest & request = dynamic_cast<Coordination::ZooKeeperGetRequest &>(*zk_request);
 
+        /// AZ node information is a special case not persisted in the storage, so we handle it first.
+        if (request.path == Coordination::keeper_availability_zone_path)
```

Comment:
Reading more, I realized should be done similar to `api_features`. insert availability_zone to `system_nodes_with_data`, https://github.com/ClickHouse/ClickHouse/pull/50796/files#diff-ebcd9a5d7e5623abd797fbfb5a7ce3cec16a9101660c1b6594c61c72d1ba015fR22

This way, no special handling here, other request like `list` would also work. Will do that in this PR & add a test.

Reply:
done.